### DISCUSSION TITLE
The code execution cannot proceed because libtcod.dll was not found

### DISCUSSION THREAD

**Discussion by Didgety** [libtcod/libtcod#151] [Q&A] [answered]
After following the tutorial: https://libtcod.readthedocs.io/en/latest/guides/getting-started.html

When I run the program it immediately crashes, running in debug gives me 2 errors.

`The code execution cannot proceed because libtcod.dll was not found`
`The code execution cannot proceed because sdl2d.dll was not found`

I see these files in their respective `./vcpkg/packages/.../bin/` folder

The example project with emscripten doesn't have this issue, but I can't figure out what is different and I don't need all that extra functionality.

My CMakeLists.txt has one additional line 
`include("${PROJECT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake")`
because the vcpkg cmake file wasn't being found without it.

Appreciate the help 🙏

**Comment by HexDecimal** [comment, marked as answer]
They're dynamically linked, so you need to copy them together with the executable, or at least have them in the working directory. The template likes to use the provider from VSCode's CMake Tools plugin to run the executable without having to move it while also keeping the project root folder as the working directory.

The files should've been moved to your CMake's bin folder automatically, or at least that's how it works from my experience.

> My CMakeLists.txt has one additional line
include("${PROJECT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake")
because the vcpkg cmake file wasn't being found without it.

It being set to `CMAKE_TOOLCHAIN_FILE` is very important. If the line setting `CMAKE_TOOLCHAIN_FILE` isn't working then you should focus on fixing that first. This could cause problems which lead to this scenario.

**Reply by Didgety** [reply]
Thank you so much for the pointer! I did some more research about vcpkg and CMAKE_TOOLCHAIN_FILE and that was exactly the problem. However, there was  a quirk that might help others in the future.

I am using CLion for my IDE on this project and the cached build files in cmake-build-debug were preventing the CMAKE_TOOLCHAIN_FILE from actually being used even though it was correctly set in CMakeLists.txt. Deleting all the build files fixed the issue and the CMAKE_TOOLCHAIN_FILE was correctly used to link the libraries.
